src/storage/core.rs

```diff
@@ -41,6 +41,7 @@ where
 {
     inner: Arc<Inner<K>>,
     observer: Observer<K>,
+    max_dirty_bytes_before_sync: u64
```

Comment:
There is no need to store that field. Value is available from config stored within `inner`